### ISSUE TITLE
Introduce "booting" status?

### ISSUE THREAD

**Issue by rikva** [hashicorp/vagrant#2182]
Hi! We are using Vagrant with the LXC provider in our integrated testsuite. Our tests consist of booting a few instances, running tests on them, and destroying the instances.
Our testsuite determines which instance to use by looking at the "vagrant status" output and booting the first available instance. 

We would like to speed up our testsuite by booting the needed instances in parallel. 
However, shortly after booting an instance, the status of this instance is still "not created". 
This results in our testsuite trying to boot identical instances simultaneously.

Wouldn't it be nice to have a "booting" status of some sorts to indicate that an instance is being booted?
